.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,55 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Pull Request Prerequisite Checks
+
+on:
+  pull_request:
+    branches:
+      - main
+  
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x]
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          fetch-depth: 0
+
+      - name: Derive appropriate SHAs for base and head for `nx affected` commands
+        uses: nrwl/nx-set-shas@v2
+        with:
+          main-branch-name: ${{ github.base_ref }}
+
+      - name: Setup Node with version ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: curl -s -u"${{ secrets.MAVEN_USERNAME }}:${{ secrets.MAVEN_PASSWORD }}" https://repo.backbase.com/api/npm/npm-backbase/auth/backbase >> .npmrc
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build app
+        run: npm run affected:build
+
+      - name: Check Formatting
+        run: npm run format:check  
+
+      - name: Unit test
+        run: npm run affected:test
+
+      - name: Check Linting
+        run: npm run affected:lint 
+      
+      - name: Install Playwright
+        run: npx playwright install --with-deps
```

Comment:
I believe this should be `npx playwright install-deps`, as the config files should already be generated

Reply:
As I understand, `playwright install-deps` installs only system dependencies for browsers, but `playwright install` will install supported browsers. There is a playwright container that can be used to run workflow steps - https://playwright.dev/docs/ci#github-actions-via-containers. With it, we can skip this step.

Also, this step takes about 1 minute, which is also a good argument to get rid of it.

![image](https://user-images.githubusercontent.com/1961590/186111535-dd9d5245-1be4-42d8-8587-5eb1e9b58f12.png)

The only thing I'm not sure about is which node version is installed in the playwright container.

Reply:
yeah that's what I was afraid of. We can optimize later, but this is a good low-hanging fruit...

Reply:
I have added the e2e tests in a separate workflow and it works fine in parallel. We can optimise this late on as well.